sbi/inference/posteriors/direct_posterior.py

```diff
@@ -101,7 +101,18 @@ def sample(
         """
 
         num_samples = torch.Size(sample_shape).numel()
+        condition_shape = self.posterior_estimator._condition_shape
```

Comment:
this could entirely replace the `x_shape` input argument unused, right? This might be another refactor, so feel free to ignore this for now but please create an issue.

Reply:
Added it #995